app/controllers/v0/profile/connected_applications_controller.rb

```diff
@@ -4,81 +4,33 @@ module V0
   module Profile
     class ConnectedApplicationsController < ApplicationController
       def index
-        render json: { data: grants_by_app.values }
+        render json: apps_from_grants, each_serializer: OktaAppSerializer
       end
 
       def destroy
-        app = grants_by_app(false)[connected_accounts_params[:id]]
-        app[:attributes][:grants].each do |grant|
-          delete_response = okta_service.delete_grant(get_user_id, grant[:id])
-          unless delete_response.success?
-            log_message_to_sentry("Error deleting grant #{connected_accounts_params[:id]}", :error,
-                                  body: delete_response.body)
-            raise 'Unable to delete grant'
-          end
-        end
+        app = OktaRedis::App.with_id(connected_accounts_params[:id])
+        app.user = @current_user
+
+        app.delete_grants
 
         head :no_content
       end
 
       private
 
-      def okta_service
-        @okta_service ||= Okta::Service.new
-      end
-
-      def get_logo(app_id)
-        app_response = okta_service.app(app_id)
-        if app_response.success?
-          app_response.body['_links']['logo'].last['href']
-        else
-          log_message_to_sentry('Error fetching app', :error,
-                                body: app_response.body)
-          nil
-        end
-      end
-
-      def apps_from_grants(grants, with_logos = true)
+      def apps_from_grants
         apps = {}
-        grants.body.each do |grant|
+        @current_user.okta_grants.all.each do |grant|
           links = grant['_links']
-          app_href = links['app']['href']
-          app_id = app_href.split('/').last
-
+          app_id = links['app']['href'].split('/').last
```

Comment:
What happens if links is nil or an empty hash?

Reply:
So these are links for a specific grant. I _think_ it's safe to assume that a grant will have a link to a specific app. If it doesn't I think having it error is probably the right call (because it's probably some weird orphaned data), though I suppose we could be more deliberate about it... what do you think?